### PR TITLE
Switch to Mojang Mappings

### PR DESCRIPTION
**Pull request by lucsoft**
Fabric and Forge both Support Mojang offical mappings

to reduce unnessary code changes, we both should use the same mappings

(Please don't squash this pr when merging as i would "break" my git history)

### CODE REVIEW

File: src/main/java/li/cil/oc2/client/gui/ComputerTerminalScreen.java [hunk 1]
```diff
@@ -22,28 +22,29 @@
...
     @Override
-    protected void drawGuiContainerBackgroundLayer(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
+    protected void renderBg(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
         terminalWidget.renderBackground(matrixStack, mouseX, mouseY);
     }
 
     @Override
-    protected void drawGuiContainerForegroundLayer(final MatrixStack matrixStack, final int mouseX, final int mouseY) {
+    protected void renderLabels(final MatrixStack p_230451_1_, final int p_230451_2_, final int p_230451_3_) {
+        // This is required to prevent the labels from beeing rendered
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/client/gui/ComputerTerminalScreen.java]: Typo, `beeing` -> `being`.

File: gradlew [hunk 2]
```diff
@@ -44,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (fnuecke) [inline gradlew]: Pretty sure I remember pipeline builds failing due to this in the past, please undo.

File: gradlew.bat [hunk 3]
```diff
@@ -33,7 +33,7 @@ set APP_HOME=%DIRNAME%
 for %%i in ("%APP_HOME%") do set APP_HOME=%%~fi
 
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=
+set DEFAULT_JVM_OPTS="-Xmx64m" "-Xms64m"
```
Comment (fnuecke) [inline gradlew.bat]: Pretty sure I remember pipeline builds failing due to this in the past, please undo.

File: Mojang.MD [hunk 4]
```diff
@@ -0,0 +1,21 @@
+**Last Modified**: 02/17/2021
```
Comment (fnuecke) [inline Mojang.MD]: This is just a disclaimer Forge generates, presumably, and can be removed?

File: src/main/java/li/cil/oc2/client/gui/RobotTerminalScreen.java [hunk 5]
```diff
@@ -28,29 +28,30 @@
...
-    protected void drawGuiContainerBackgroundLayer(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
-        INVENTORY_BACKGROUND.draw(matrixStack, guiLeft + SLOTS_X, guiTop + SLOTS_Y);
+    protected void renderBg(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
+        INVENTORY_BACKGROUND.draw(matrixStack, leftPos + SLOTS_X, topPos + SLOTS_Y);
         terminalWidget.renderBackground(matrixStack, mouseX, mouseY);
     }
 
     @Override
-    protected void drawGuiContainerForegroundLayer(final MatrixStack matrixStack, final int mouseX, final int mouseY) {
+    protected void renderLabels(final MatrixStack p_230451_1_, final int p_230451_2_, final int p_230451_3_) {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/client/gui/RobotTerminalScreen.java]: Please keep the readable parameter names.

File: src/main/java/li/cil/oc2/client/gui/ComputerTerminalScreen.java [hunk 6]
```diff
@@ -22,28 +22,29 @@
...
 
     @Override
-    protected void drawGuiContainerBackgroundLayer(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
+    protected void renderBg(final MatrixStack matrixStack, final float partialTicks, final int mouseX, final int mouseY) {
         terminalWidget.renderBackground(matrixStack, mouseX, mouseY);
     }
 
     @Override
-    protected void drawGuiContainerForegroundLayer(final MatrixStack matrixStack, final int mouseX, final int mouseY) {
+    protected void renderLabels(final MatrixStack p_230451_1_, final int p_230451_2_, final int p_230451_3_) {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/client/gui/ComputerTerminalScreen.java]: Please keep the readable parameter names.

File: src/main/java/li/cil/oc2/common/entity/RobotEntity.java [hunk 7]
```diff
@@ -332,21 +331,16 @@ public boolean canBeCollidedWith() {
...
         return entity != this;
     }
 
     @Override
-    public void applyEntityCollision(final Entity entity) {
+    public void push(final Entity entity) {
     }
 
     @Override
-    public boolean func_241845_aY() {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/entity/RobotEntity.java:344]: Why was this removed? This is what allows players to stand on top of robots.

File: src/main/java/li/cil/oc2/common/entity/RobotEntity.java [hunk 8]
```diff
@@ -489,12 +484,10 @@ public void set(final int index, final int value) {
                     }
 
                     @Override
-                    public int size() {
-                        return 3;
-                    }
+                    public int getCount() { return 3; }
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/entity/RobotEntity.java]: Please keep the code style, so the return on a new line here.

File: src/main/java/li/cil/oc2/common/tileentity/AbstractTileEntity.java [hunk 9]
```diff
@@ -96,8 +95,8 @@ public void onWorldUnloaded() {
     }
 
     @Override
-    public void remove() {
-        super.remove(); // -> invalidateCaps()
+    public void setRemoved() {
+        super.setRemoved();
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/tileentity/AbstractTileEntity.java]: Please keep that comment.

File: src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java [hunk 10]
```diff
@@ -58,49 +58,49 @@ public int getOutputForDirection(final Direction direction) {
...
         assert localDirection != null;
 
-        return output[localDirection.getIndex()];
+        return output[localDirection.get3DDataValue()];
     }
 
     @Callback(name = GET_REDSTONE_INPUT)
-    public int getRedstoneInput(@Parameter(SIDE) final Direction side) {
-        final World world = getWorld();
+    public int setSignal(@Parameter(SIDE) final Direction side) {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java]: Please keep the method names of the device API.

File: src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java [hunk 11]
```diff
@@ -58,49 +58,49 @@ public int getOutputForDirection(final Direction direction) {
...
         }
 
-        return world.getRedstonePower(neighborPos, direction);
+        return world.getSignal(neighborPos, direction);
     }
 
     @Callback(name = GET_REDSTONE_OUTPUT, synchronize = false)
-    public int getRedstoneOutput(@Parameter(SIDE) final Direction side) {
-        return output[side.getIndex()];
+    public int getSignal(@Parameter(SIDE) final Direction side) {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java]: Please keep the method names of the device API.

File: src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java [hunk 12]
```diff
@@ -58,49 +58,49 @@ public int getOutputForDirection(final Direction direction) {
...
+        return output[side.get3DDataValue()];
     }
 
     @Callback(name = SET_REDSTONE_OUTPUT)
-    public void setRedstoneOutput(@Parameter(SIDE) final Direction side, @Parameter(VALUE) final int value) {
+    public void getSignal(@Parameter(SIDE) final Direction side, @Parameter(VALUE) final int value) {
```
Comment (fnuecke) [inline src/main/java/li/cil/oc2/common/tileentity/RedstoneInterfaceTileEntity.java]: Please keep the method names of the device API.

**Review by fnuecke — CHANGES_REQUESTED** [review]

File: src/main/java/li/cil/oc2/common/entity/RobotEntity.java (hunk 7 above)
Comment (lucsoft) [reply src/main/java/li/cil/oc2/common/entity/RobotEntity.java:344]: Thanks this fixed the bug where i couldn't interact with the robot

**Review by fnuecke — APPROVED** [review]

**Comment by lucsoft** [comment]
👏 👏 👏
